msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/mam/TestCase2516967.kt

```diff
@@ -0,0 +1,120 @@
+import com.microsoft.identity.client.ui.automation.app.OutlookApp
+import com.microsoft.identity.client.ui.automation.broker.BrokerCompanyPortal
+import com.microsoft.identity.client.ui.automation.broker.BrokerHost
+import com.microsoft.identity.client.ui.automation.broker.IMdmAgent
+import com.microsoft.identity.client.ui.automation.installer.LocalApkInstaller
+import com.microsoft.identity.client.ui.automation.interaction.FirstPartyAppPromptHandlerParameters
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter
+import com.microsoft.identity.common.java.util.ThreadUtils
+import com.microsoft.identity.labapi.utilities.client.LabQuery
+import com.microsoft.identity.labapi.utilities.constants.ProtectionPolicy
+import com.microsoft.identity.labapi.utilities.constants.TempUserType
+import com.microsoft.identity.labapi.utilities.constants.UserType
+import org.junit.Assert
+import org.junit.Test
+
+// Can use Outlook with True MAM account upon re-registration
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2516967
+@SupportedBrokers(brokers = [BrokerCompanyPortal::class])
+class TestCase2516967 : AbstractMsalBrokerTest(){
+
+    @Test
+    fun test_2516967() {
+        // Fetch credentials
+        val username: String = mLabAccount.username
+        val password: String = mLabAccount.password
+
+        val outlook = OutlookApp(LocalApkInstaller())
+        outlook.install()
+        outlook.launch()
+        outlook.handleFirstRun()
+
+        val promptHandlerParameters = FirstPartyAppPromptHandlerParameters.builder()
+            .broker(mBroker)
+            .prompt(PromptParameter.SELECT_ACCOUNT)
+            .loginHint(username)
+            .consentPageExpected(false)
+            .sessionExpected(false)
+            .expectingBrokerAccountChooserActivity(false)
+            .expectingLoginPageAccountPicker(false)
+            .registerPageExpected(true)
+            .build()
+
+        // add first account in Outlook
+        outlook.addFirstAccount(username, password, promptHandlerParameters)
+        outlook.onAccountAdded()
+        // handle app protection policy in CP i.e. setup PIN when asked
+        (mBroker as IMdmAgent).handleAppProtectionPolicy()
+
+        val brokerHost = BrokerHost()
+        brokerHost.install()
+        brokerHost.wpjLeave()
+
+        // advance clock by more than an hour to expire AT in cache
+        settingsScreen.forwardDeviceTimeForOneDay()
+
+        // Log in again in outlook, should get a prompt in the snackbar
+        outlook.launch()
+        outlook.signInThroughSnackBar(username, password, promptHandlerParameters)
+
+        // Sometimes outlook takes some time to remove the SignIn refresh snackbar
+        ThreadUtils.sleepSafely(6000, "sleeping", "interrupted sleep")
+        outlook.forceStop()
+        outlook.launch()
+        outlook.forceStop()
+        outlook.launch()
```

Comment:
Why twice? Some comments here should help

Reply:
Added a comment